### PR TITLE
Allow tool form to listen to history

### PR DESCRIPTION
**Pull request by nturaga**
made changes to tool-form-base.js, this should allow the tool-form to listen to history. Thanks @guerler for walking me through backbone

### CODE REVIEW

**Comment by nturaga** [comment]
@guerler Comments were lost in rebase. Not sure why. I made one of the changes you suggested, but the goal was to get it in before the freeze tomorrow. I figured we can make rest of the changes after, since this works as is.

File: client/galaxy/scripts/mvc/tool/tool-form-base.js [hunk 1]
```diff
@@ -11,11 +11,26 @@ define(['utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view',
             this.deferred = new Deferred();
             if (options.inputs) {
                 this._buildForm(options);
+                this.refresh();
```
Comment (guerler) [inline client/galaxy/scripts/mvc/tool/tool-form-base.js]: This refresh request does not seem to be necessary.

File: client/galaxy/scripts/mvc/tool/tool-form-base.js [hunk 2]
```diff
@@ -38,9 +53,7 @@ define(['utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view',
                 operations      : !this.options.hide_operations && this._operations(),
                 onchange        : function() {
                     self.deferred.reset();
```
Comment (guerler) [inline client/galaxy/scripts/mvc/tool/tool-form-base.js]: The reset here should probably be placed into the refresh call. It removes uneccessarily queued requests.

Comment (nturaga) [inline client/galaxy/scripts/mvc/tool/tool-form-base.js]: thanks @guerler

**Comment by guerler** [comment]
+1 waiting for tests to complete.

**Pull request merged**